onnxruntime/core/providers/cpu/controlflow/scan_utils.h

```diff
@@ -62,6 +62,8 @@ class LoopStateVariable {
     */
   MLValue a_;
   MLValue b_;
+
+  bool is_copy_of_preexisting_value_;
```

Comment:
I am concerned about the interaction between the new sharing and the existing scan implementation technique of alternating between a_ and b_ for LoopStateVariables. If LoopStateVariable X is copied to LoopStateVariable Y via an identity, and we let Y "share" ml-value of X, subsequently we are obliged NOT to update X's value in-place (otherwise, Y will incorrectly see these updates). But scan's implementation reuses a_ and b_, so it could end up updating this memory, right? (It might manifest itself in 1 iteration; if not, it might in later iterations: e.g., we could have "Y_out = X_in; Z_out = Y_in".)

Reply:
The problem doesn't manifest itself for "Loop" (I think) because Loop doesn't reuse its (loop-body's) input ml-values underlying memory. The easiest fix may be not to enable it for scan. If we want to do it for scan, we could restrict it to the case where we have "X_out = X_in" (where the identity op is between same state-variable's in and out values)/